src/openrct2-ui/windows/Finances.cpp

```diff
@@ -341,1048 +260,820 @@ static constexpr const uint32_t WindowFinancesPageHoldDownWidgets[] =
 
 #pragma endregion
 
-static constexpr const int32_t WindowFinancesTabAnimationLoops[] =
-{
-    16, 32, 32, 32, 38, 16
-};
-
-static constexpr const int32_t EXPENDITURE_COLUMN_WIDTH = 80;
-
-static int32_t _lastPaintedMonth;
-
-static constexpr const rct_string_id window_finances_summary_row_labels[static_cast<int32_t>(ExpenditureType::Count)] = {
-    STR_FINANCES_SUMMARY_RIDE_CONSTRUCTION,
-    STR_FINANCES_SUMMARY_RIDE_RUNNING_COSTS,
-    STR_FINANCES_SUMMARY_LAND_PURCHASE,
-    STR_FINANCES_SUMMARY_LANDSCAPING,
-    STR_FINANCES_SUMMARY_PARK_ENTRANCE_TICKETS,
-    STR_FINANCES_SUMMARY_RIDE_TICKETS,
-    STR_FINANCES_SUMMARY_SHOP_SALES,
-    STR_FINANCES_SUMMARY_SHOP_STOCK,
-    STR_FINANCES_SUMMARY_FOOD_DRINK_SALES,
-    STR_FINANCES_SUMMARY_FOOD_DRINK_STOCK,
-    STR_FINANCES_SUMMARY_STAFF_WAGES,
-    STR_FINANCES_SUMMARY_MARKETING,
-    STR_FINANCES_SUMMARY_RESEARCH,
-    STR_FINANCES_SUMMARY_LOAN_INTEREST,
-};
 // clang-format on
 
-static void WindowFinancesSetPage(rct_window* w, int32_t page);
-static void WindowFinancesSetPressedTab(rct_window* w);
-static void WindowFinancesDrawTabImages(rct_drawpixelinfo* dpi, rct_window* w);
-
-/**
- *
- *  rct2: 0x0069DDF1
- */
-rct_window* WindowFinancesOpen()
+class FinancesWindow final : public Window
 {
-    rct_window* w;
+private:
+    int32_t _lastPaintedMonth;
 
-    w = window_bring_to_front_by_class(WC_FINANCES);
-    if (w == nullptr)
+public:
+    void OnOpen() override
     {
-        w = WindowCreateAutoPos(WW_OTHER_TABS, WH_SUMMARY, _windowFinancesPageEvents[0], WC_FINANCES, WF_10);
-        w->number = 0;
-        w->frame_no = 0;
-
+        SetPage(WINDOW_FINANCES_PAGE_SUMMARY);
```

Comment:
_lastPaintedMonth should be set to `std::numeric_limits<int32_t>::max()` to force it to be recalculated on first draw.